packages/web3-providers/src/rarible/index.ts

```diff
@@ -135,9 +167,212 @@ export async function getHistoryFromRarible(tokenAddress: string, tokenId: strin
         const ownerInfo = profiles.find((profile) => profile.id === history.owner)
         const fromInfo = profiles.find((profile) => profile.id === history.buyer || profile.id === history.from)
         return {
-            ...history,
-            ownerInfo,
-            fromInfo,
-        }
+            id: history.id,
+            eventType: history['@type'],
+            timestamp: history.date.getTime() ?? 0,
+            price: {
+                quantity: '1',
+                price: history.price,
+                asset: {
+                    id: fromInfo?.id,
+                    decimals: 0,
+                    image_url: fromInfo?.image,
+                    image_original_url: '',
+                    image_preview_url: '',
+                    asset_contract: {
+                        symbol: fromInfo?.type,
+                    },
+                    permalink: '',
+                },
+            },
+            accountPair: {
+                from: {
+                    username: fromInfo?.name,
+                    address: fromInfo?.id,
+                    imageUrl: fromInfo?.image,
+                    link: '',
+                },
+                to: {
+                    username: ownerInfo?.name,
+                    address: ownerInfo?.id,
+                    imageUrl: ownerInfo?.image,
+                    link: '',
+                },
+            },
+        } as NFTHistory
+    })
+}
+
+//#region new code
```

Comment:
We don't add a title at the section end.

```suggestion
//#region
```